app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginViewModel.kt

```diff
@@ -2,37 +2,144 @@ package com.realityexpander.tasky.presentation.login_screen
 
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.realityexpander.tasky.common.Exceptions
-import com.realityexpander.tasky.data.repository.AuthRepositoryImpl
-import com.realityexpander.tasky.data.repository.local.AuthApiImpl
-import com.realityexpander.tasky.data.repository.remote.AuthDaoImpl
-import com.realityexpander.tasky.data.validation.ValidateEmailImpl
 import com.realityexpander.tasky.domain.IAuthRepository
+import com.realityexpander.tasky.domain.validation.IValidateEmail
+import com.realityexpander.tasky.domain.validation.IValidatePassword
+import com.realityexpander.tasky.presentation.common.UIConstants.SAVED_STATE_EMAIL
+import com.realityexpander.tasky.presentation.common.UIConstants.SAVED_STATE_PASSWORD
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class LoginViewModel @Inject constructor(
-    private val authRepository: IAuthRepository = AuthRepositoryImpl(
-        authApi = AuthApiImpl(),
-        authDao = AuthDaoImpl(),
-        validateEmail = ValidateEmailImpl()
-    ),
+    private val authRepository: IAuthRepository,
+    private val validateEmail: IValidateEmail,
+    private val validatePassword: IValidatePassword,
     private val savedStateHandle: SavedStateHandle,
 ) : ViewModel() {
 
-    suspend fun login(email: String, password: String) {
+    private val _loginState = MutableStateFlow<LoginState>(LoginState())
+    val loginState: StateFlow<LoginState> = _loginState.asStateFlow()
+
+    private val email: String = savedStateHandle[SAVED_STATE_EMAIL] ?: ""
+    private val password: String = savedStateHandle[SAVED_STATE_PASSWORD] ?: ""
```

Comment:
Why are these the only fields you want to restore after process death? What about validation errors or status messages for example?

Reply:
Good point, i will include those fields for process death. 🧟‍♂️